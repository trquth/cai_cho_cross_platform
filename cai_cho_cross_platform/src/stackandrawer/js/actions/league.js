import { GET_LEAGUE } from './types';
import { getTeams } from '../db/helpers';

export function getLeague(currentLeagueRoute) {
  const teams = getTeams(currentLeagueRoute);

  return {
    type: GET_LEAGUE,
    payload: { teams },
  };
}
