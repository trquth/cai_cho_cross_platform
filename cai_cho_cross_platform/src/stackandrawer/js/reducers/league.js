import { GET_LEAGUE } from '../actions/types';
import { getTeams } from '../db/helpers';

const initialState = {
  teams: getTeams('PremierLeague'),
};

export default function (state = initialState, action) {
  if (action.type === GET_LEAGUE) {
    return { ...action.payload };
  }

  return state;
}
