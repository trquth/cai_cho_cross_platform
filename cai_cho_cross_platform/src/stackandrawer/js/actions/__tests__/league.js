import mockStore from 'redux-mock-store';
import * as actions from '../league';

const store = mockStore();

beforeEach(() => {
  store.clearActions();
});

it('GET_LEAGUE (LaLiga)', () => {
  store.dispatch(actions.getLeague('LaLiga'));
  expect(store.getActions()).toMatchSnapshot();
});
