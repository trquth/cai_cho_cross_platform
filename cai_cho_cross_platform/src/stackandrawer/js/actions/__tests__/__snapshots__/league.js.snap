// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET_LEAGUE (LaLiga) 1`] = `
Array [
  Object {
    "payload": Object {
      "teams": Object {
        "clubs": Array [
          Object {
            "code": "BAR",
            "key": "barcelona",
            "name": "Barcelona",
          },
          Object {
            "code": "ESP",
            "key": "espanyol",
            "name": "Espanyol",
          },
          Object {
            "code": "RMD",
            "key": "madrid",
            "name": "R. Madrid",
          },
          Object {
            "code": "ATL",
            "key": "atletico",
            "name": "Atlético",
          },
          Object {
            "code": "SEV",
            "key": "sevilla",
            "name": "Sevilla",
          },
          Object {
            "code": "VAL",
            "key": "valencia",
            "name": "Valencia",
          },
          Object {
            "code": "MAG",
            "key": "malaga",
            "name": "Málaga",
          },
          Object {
            "code": "ATH",
            "key": "athletic",
            "name": "Athletic",
          },
          Object {
            "code": "GRA",
            "key": "granada",
            "name": "Granada",
          },
          Object {
            "code": "CEL",
            "key": "celta",
            "name": "Celta",
          },
          Object {
            "code": "RSO",
            "key": "realsociedad",
            "name": "R. Sociedad",
          },
          Object {
            "code": "DEP",
            "key": "deportivo",
            "name": "Deportivo",
          },
          Object {
            "code": null,
            "key": "eibar",
            "name": "Eibar",
          },
          Object {
            "code": "BET",
            "key": "betis",
            "name": "Betis",
          },
          Object {
            "code": null,
            "key": "sporting",
            "name": "Sporting",
          },
          Object {
            "code": null,
            "key": "laspalmas",
            "name": "Las Palmas",
          },
          Object {
            "code": "OSA",
            "key": "osasuna",
            "name": "Osasuna",
          },
          Object {
            "code": "VLL",
            "key": "villareal",
            "name": "Villarreal CF",
          },
          Object {
            "code": null,
            "key": "alaves",
            "name": "Alavés",
          },
          Object {
            "code": null,
            "key": "leganes",
            "name": "Club Deportivo Leganés",
          },
        ],
        "name": "Primera División 2016/17",
        "routeName": "LaLiga",
      },
    },
    "type": "GET_LEAGUE",
  },
]
`;
